### PR TITLE
[Feature] 10 digit teletan with checksum check

### PR DESCRIPTION
**Pull request by thomasaugsten**
- Add 10 digit teletan with checksum and forbidden character check

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Views/BaseElements/ENATanInput.swift [hunk 1]
```diff
@@ -104,27 +130,98 @@ class ENATanInput: UIControl, UIKeyInput {
...
+			return false
+
+		}
+		let start = input.index(input.startIndex, offsetBy: 0)
+		let end = input.index(input.startIndex, offsetBy: input.count - 2)
+		let testString = String(input[start...end])
+		return input.last == calculateChecksum(input: testString)
+	}
+	
+	func calculateChecksum(input: String) -> Character {
```
Comment (EisvogelSAP) [inline src/xcode/ENA/ENA/Source/Views/BaseElements/ENATanInput.swift:184]: Semantically this should be handled in the corresponding controller. But is fine for now.

File: src/xcode/ENA/ENA/Source/Views/BaseElements/ENATanInput.swift [hunk 2]
```diff
@@ -19,32 +19,41 @@ import Foundation
...
 class ENATanInput: UIControl, UIKeyInput {
-	@IBInspectable var textColor: UIColor = UIColor.preferredColor(for: .textPrimary1)
-	@IBInspectable var boxColor: UIColor = UIColor.preferredColor(for: .backgroundSecondary)
+	@IBInspectable var labelTextColor: UIColor = .enaColor(for: .textPrimary1)
+	@IBInspectable var boxColor: UIColor = .enaColor(for: .separator)
 
 	@IBInspectable var fontSize: CGFloat = 30
-	@IBInspectable var digits: Int = 7
+	@IBInspectable var groups: String = "3,3,4"
+	@IBInspectable var forbidden: String = "1,0,I,O,L,Ä,Ö,Ü, ,-,/,:,;,(,),€,&,@,\""
```
Comment (EisvogelSAP) [inline src/xcode/ENA/ENA/Source/Views/BaseElements/ENATanInput.swift:32]: What about all other characters? The whole world of unicode is open. Should this rather be a whitelist character set?

**Review by eisvogel — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionTanInputViewController.swift [hunk 3]
```diff
@@ -109,8 +110,19 @@ extension ExposureSubmissionTanInputViewController: ExposureSubmissionNavigation
...
 
-	func tanChanged(isValid: Bool) {
-		setButtonEnabled(enabled: isValid)
+	func tanChanged(isValid: Bool, checksumIsValid: Bool, isBlocked: Bool) {
+		setButtonEnabled(enabled: (isValid && checksumIsValid))
+		if (isValid && !checksumIsValid) {
+			errorLabel.text = AppStrings.ExposureSubmissionTanEntry.invalidError
+			errorLabel.isHidden = false
+		}
+		else if (isBlocked) {
```
Comment (johannesrohwer) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionTanInputViewController.swift:119]: does the linter allow () in conditional statements?

**Review by johannesrohwer — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionTanInputViewController.swift (hunk 3 above)
Comment (melloskitten) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionTanInputViewController.swift:119]: remove () around condition

**Review by ChristopherSchmitz — APPROVED** [review]

**Pull request merged**